pcdsdevices/utils.py

```diff
@@ -527,15 +548,16 @@ def set_many(
     """
     statuses = []
     log = owner.log if owner is not None else logger
+    set_kw = dict(timeout=timeout, settle_time=settle_time)
     for signal, value in to_set.items():
+        # Decide which keywords are suitable for this signal
+        params = inspect.signature(signal.set).parameters
+        _kw = {key: val for key, val in set_kw.items() if key in params.keys()}
```

Comment:
The test suite is failing because this approach breaks the kwargs-passing from this function for any signals who accept `timeout` and `settle_time` via `**kwargs`. Since the parameters aren't explicitly in the signature they get skipped, and the test that checks that they've been called with the given arguments fails.

Reply:
Ah yes. Okay, I've nested the ``set()`` in another ``try…except TypeError`` block that retries the ``set()`` call without the *settle_time* argument. I don't really love the nested ``try…except``s, but I think this describes what I'm actually trying to do.

I didn't include it as another ``except TypeError:`` for the outer ``try…except`` block because if the second attempt also raised a ``TypeError``, then this would bypass the logger in ``except Exception:``.